tests/v2/integration/catalogv2/cluster_repo_test.go

```diff
@@ -134,14 +137,30 @@ func (c *ClusterRepoTestSuite) TestGitRepo() {
 	})
 }
 
-func StartRegistry() (*url.URL, error) {
-	s := httptest.NewServer(registryGoogle.New())
-	u, err := url.Parse(s.URL)
+func StartRegistry() (*httptest.Server, error) {
+	// Create a new registry handler
+	handler := registryGoogle.New()
+
+	// Optionally, you can customize the handler here if needed
+	// e.g., add middleware, logging, etc.
+	customHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		logrus.Infof("Received request: %s %s", r.Method, r.URL)
+		handler.ServeHTTP(w, r)
+	})
+
+	// Create a new test server
+	ts := httptest.NewUnstartedServer(customHandler)
+
+	ip := getOutboundIP()
+	// Bind the server to a specific IP address (your local machine's IP)
+	listener, err := net.Listen("tcp", fmt.Sprintf("%s:4050", ip.String()))
 	if err != nil {
-		return nil, err
+		log.Fatalf("Failed to bind to local IP: %v", err)
```

Comment:
My mistake can you change this to  return nil, err instead of a fatal?

Reply:
done